packages/grafana-traces/src/utils.ts

```diff
@@ -37,7 +37,7 @@ export function getNonOverlappingDuration(ranges: Array<[number, number]>): numb
  * both arrays and dataframe views.
  */
 export function makeSpanMap<T>(getSpan: (index: number) => { span: T; id: string; parentIds: string[] } | undefined): {
-  [id: string]: { span: T; children: string[] };
+  [id: string]: { span?: T; children: string[] };
```

Comment:
Don't remember writing this, seems like it may be something from jeager codebase but I would say this should actually return `[id: string]: { span: T; children: string[] }`.

The thing I assume is that it is creating a tree with children from a list of spans with parentIds. Those spans may not be in order of tree tarversal and so we can get child span before we get parent span. In that case we create a temporary span object to have a placeholder for the ID and fill in the span data later. We could recheck in the end to make sure and maybe just delete anything that is still `span: undefined` but even in that case I would say the type assertion will be necessary.